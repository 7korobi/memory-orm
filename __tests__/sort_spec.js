/*
 * decaffeinate suggestions:
 * DS102: Remove unnecessary code created because of implicit returns
 * Full docs: https://github.com/decaffeinate/decaffeinate/blob/master/docs/suggestions.md
 */
require('./models/index')
const Mem = require('../lib/index')

const keys = [
  'tags',
  'faces',
  'chr_sets',
  'chr_npcs',
  'chr_jobs',
  'locales',
  'randoms',
  'folders',
  'roles',
  'ables',
  'sow_roletables',
]

keys.map((key) =>
  describe(key, function () {
    test('sort snapshot', () => expect(Mem.Query[key].$sort).toMatchSnapshot())
  })
)
