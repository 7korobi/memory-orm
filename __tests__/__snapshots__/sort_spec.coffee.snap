// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ables sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`chr_jobs sort snapshot 1`] = `
Object {
  "_reduce.list": Object {
    "sort": Array [
      "face.order",
    ],
  },
}
`;

exports[`chr_npcs sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`chr_sets sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`faces sort snapshot 1`] = `
Object {
  "_reduce.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.name_head": Object {
    "cover": Array [
      "ア",
      "ィ",
      "イ",
      "ゥ",
      "ウ",
      "ェ",
      "エ",
      "ォ",
      "オ",
      "カ",
      "ガ",
      "キ",
      "ギ",
      "ク",
      "グ",
      "ケ",
      "ゲ",
      "コ",
      "ゴ",
      "サ",
      "ザ",
      "シ",
      "ジ",
      "ス",
      "ズ",
      "セ",
      "ゼ",
      "ソ",
      "ゾ",
      "タ",
      "ダ",
      "チ",
      "ヂ",
      "ッ",
      "ツ",
      "ヅ",
      "テ",
      "デ",
      "ト",
      "ド",
      "ナ",
      "ニ",
      "ヌ",
      "ネ",
      "ノ",
      "ハ",
      "バ",
      "パ",
      "ヒ",
      "ビ",
      "ピ",
      "フ",
      "ブ",
      "プ",
      "ヘ",
      "ベ",
      "ペ",
      "ホ",
      "ボ",
      "ポ",
      "マ",
      "ミ",
      "ム",
      "メ",
      "モ",
      "ャ",
      "ヤ",
      "ュ",
      "ユ",
      "ョ",
      "ヨ",
      "ラ",
      "リ",
      "ル",
      "レ",
      "ロ",
      "ヮ",
      "ワ",
      "ヰ",
      "ヱ",
      "ヲ",
      "ン",
    ],
    "index": "set.length",
    "sort": Array [
      "id",
    ],
  },
}
`;

exports[`folders sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`locales sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`randoms sort snapshot 1`] = `
Object {
  "_reduce.list": Object {
    "sort": Array [
      "ratio",
      "desc",
    ],
  },
  "_reduce.type": Object {
    "sort": Array [
      "count",
      "asc",
    ],
  },
}
`;

exports[`roles sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`sow_roletables sort snapshot 1`] = `
Object {
  "_reduce.list": Object {},
}
`;

exports[`tags sort snapshot 1`] = `
Object {
  "_reduce.group.1.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.10.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.2.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.4.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.5.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.6.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.7.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.8.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.group.9.list": Object {
    "sort": Array [
      "order",
    ],
  },
  "_reduce.list": Object {
    "sort": Array [
      "order",
    ],
  },
}
`;
